In [2]:
import pandas as pd
import pandasql
from pandasql import sqldf

In [5]:
!pip install openpyxl

     -------------------------------------- 242.1/242.1 kB 3.7 MB/s eta 0:00:00
  Using cached et_xmlfile-1.1.0-py3-none-any.whl (4.7 kB)


In [6]:
# create df from xlsx file
df = pd.read_excel('imdb_movies.xlsx')
df.head()

ID Language                        Title  Popularity Release_Date  \
0  580489       en  Venom: Let There Be Carnage    5401.308   2021-09-30   
1  524434       en                     Eternals    3365.535   2021-11-03   
2  438631       en                         Dune    2911.423   2021-09-15   
3  796499       en              Army of Thieves    2552.437   2021-10-27   
4  550988       en                     Free Guy    1850.470   2021-08-11   

   Vote_Average  Vote_Count  \
0           6.8        1736   
1           7.1         622   
2           8.0        3632   
3           6.9         555   
4           7.8        3493   

                                               Genre      Revenue  Runtime  
0         ['Science Fiction', 'Action', 'Adventure']  424000000.0     97.0  
1  ['Action', 'Adventure', 'Science Fiction', 'Fa...  165000000.0    157.0  
2         ['Action', 'Adventure', 'Science Fiction']  331116356.0    155.0  
3                    ['Action', 'Crime', 'Thriller']          0.0    127.0  
4  ['Comedy', 'Action', 'Adventure', 'Science Fic...  331096766.0    115.0

In [7]:
# check dtypes of df before querying
df.dtypes

ID                       int64
Language                object
Title                   object
Popularity             float64
Release_Date    datetime64[ns]
Vote_Average           float64
Vote_Count               int64
Genre                   object
Revenue                float64
Runtime                float64
dtype: object

#  Does highest popularity equal an overall highest earned revenue film? (i.e. top ranked popular movie(s)  = highest earned revenue films)

In [10]:
# query to find max popularity film
df1 = pandasql.sqldf('SELECT "Title", MAX("Popularity") as Popularity, "Release_Date", "Genre", "Revenue" FROM df', globals())
df1.head()

Title  Popularity                Release_Date  \
0  Venom: Let There Be Carnage    5401.308  2021-09-30 00:00:00.000000   

                                        Genre      Revenue  
0  ['Science Fiction', 'Action', 'Adventure']  424000000.0

In [11]:
# query to find max revenue film
df1 = pandasql.sqldf('SELECT "Title", "Popularity", "Release_Date", "Genre", MAX("Revenue") as Revenue FROM df', globals())
df1.head()

Title  Popularity                Release_Date  \
0  Avatar     374.059  2009-12-10 00:00:00.000000   

                                               Genre       Revenue  
0  ['Action', 'Adventure', 'Fantasy', 'Science Fi...  2.847246e+09

### Avatar generated 2.85 billion dollars in revenue; however, Venom: Let There Be Carnage generated only 424 million in revenue. Avatar's popularity score only 374. The higher the popularity score, the more popular the movie. We can conclude that highest popularity scored film does not equal highest revenue film.

#  Which movie ranked highest vote count?

In [29]:
df1 = pandasql.sqldf('SELECT "Title", "Popularity", "Release_Date", "Genre", MAX("Vote_Count") as Vote FROM df', globals())
df1.head()

Title  Popularity                Release_Date  \
0  Inception     144.741  2010-07-15 00:00:00.000000   

                                        Genre   Vote  
0  ['Action', 'Science Fiction', 'Adventure']  30184

### Inception movie was determined to highest voted movie by users on IMDB website. This could change due to the dataset not being up to date as IMDB webpage. When a user submits a vote for a movie the vote is not automatically updated but instead updated numerous times a day according to IMDB FAQ's [https://help.imdb.com/article/imdb/track-movies-tv/ratings-faq/G67Y87TFYYP6TWAV#]

In [30]:
df1 = pandasql.sqldf('SELECT "Title", "Popularity", "Release_Date", "Genre", "Vote_Count" FROM df ORDER BY "Release_Date" DESC', globals())
df1.head()

Title  Popularity  \
0                    Fast & Furious 10     211.849   
1                             Avatar 2      58.466   
2                            The Flash      70.092   
3                       Halloween Ends      32.685   
4  Spider-Man: Across the Spider-Verse      75.702   

                 Release_Date  \
0  2023-04-06 00:00:00.000000   
1  2022-12-14 00:00:00.000000   
2  2022-11-02 00:00:00.000000   
3  2022-10-13 00:00:00.000000   
4  2022-10-06 00:00:00.000000   

                                               Genre  Vote_Count  
0                                ['Action', 'Crime']           0  
1  ['Action', 'Adventure', 'Science Fiction', 'Fa...           0  
2  ['Action', 'Adventure', 'Science Fiction', 'Fa...           0  
3                             ['Horror', 'Thriller']           0  
4  ['Animation', 'Action', 'Science Fiction', 'Co...           0